libphal/libphal.H

```diff
@@ -278,6 +280,44 @@ bool isSbeVitalAttnActive(struct pdbg_target *proc);
  */
 bool hasControlTransitionedToHost();
 
+using LocationCode = std::string;
+using FRUType = uint8_t;
+constexpr uint8_t EXP_LOCATIN_CODE_MIN_LENGTH = 17;
+constexpr uint8_t UNEXP_LOCATIN_CODE_MIN_LENGTH = 4;
+
+/**
+ * @brief Get unexpanded location code
+ *
+ * @details An api to get an unexpanded location code corresponding to
+ *          a given expanded location code. Unexpanded location codes
+ *          give the location of the FRU in the system.
+ *
+ * @param[in] locCode - Location code in expanded format.
+ *
+ * @return Location code as string which will be in un-expanded format or
+ *         empty optional on failure.
```

Comment:
this is a small code which just checks if it has the value, do we need to use different error codes and complicate why can;t the user simply check the std::nullopt and take decision

Reply:
If a user is wants to create a error log (aka PEL) or needs to take action based on an error, how can they do so?

There are a few scenarios:

If the user provides an incorrect location code format, how will the API user know? The API verifies the location code format to get the unexpanded format location code.

If the user provides a valid location code, but it not matches a predefined FRU list, how can the PHAI API bug be detected?

Therefore, I recommend returning an error code regardless of whether the caller uses it or not, as it does not impact the API implementation.

If you prefer using exceptions as in the past, that's acceptable. However, I generally do not recommend it, as it requires writing try and catch blocks in every caller code. Instead, I suggest using std::expected. This approach aligns with the existing PHAL infrastructure, which provides clear error codes and messages. The caller can then use these messages for tracing purposes, taking different actions based on the error code, or ignoring the error codes if they are not relevant. I believe this will be more convenient for the caller and will also help in detecting API implementation bugs.